transport/tlscommon/tls.go

```diff
@@ -113,29 +117,24 @@ func ReadPEMFile(log *logp.Logger, s, passphrase string) ([]byte, error) {
 			break
 		}
 
-		if x509.IsEncryptedPEMBlock(block) { //nolint: staticcheck // deprecated, we have to get rid of it
-			var buffer []byte
-			var err error
-			if len(pass) == 0 {
-				err = errors.New("no passphrase available")
-			} else {
-				// Note, decrypting pem might succeed even with wrong password, but
-				// only noise will be stored in buffer in this case.
-				buffer, err = x509.DecryptPEMBlock(block, pass) //nolint: staticcheck // deprecated, we have to get rid of it
+		switch {
+		case x509.IsEncryptedPEMBlock(block): //nolint: staticcheck // deprecated, we have to get rid of it
+			block, err := decryptPKCS1Key(*block, pass)
+			if err != nil {
+				log.Errorf("Dropping encrypted pem with private key '%v': %v", block.Type, err)
```

Comment:
I'm a bit confused by this error message, on a first glance I'd expect the private key to be logged as the first argument :scream: , but because the value is `block.Type` I believe it the message should better reflect what it means, something like that:
```suggestion
				log.Errorf("Dropping encrypted pem block with private key, block type '%s': %s", block.Type, err)
```

Also `block.Type` is a string, so `%s` is more appropriated there. For the error I also rather use `%s`, but that's totally up to you.